### DISCUSSION TITLE
Has anybody succeded running keycloak quarkus 18 or  19 docker behind nginx proxy and ssl offloading?

### DISCUSSION THREAD

**Discussion by Ali-Javanmardi** [keycloak/keycloak#13651] [Q&A] [answered]
Hi,

This is a few days I'm trying to run a keycloak quarkus 18 or  19 server behind an nginx proxy server. (both running docker containers)
I have tried any setting described in main keycloack documentation or found in any blog or issue in keycloak github page.

Unfortunately the only version I could run behind proxy and had access to admin page was version 17.

I have tried "edge, reencrypt, passthrough" modes one by one and tried every settings related to proxy as I found in documentation.
I have access to the root path of keycloak but as soon as I try to access administration console I receive an empty page.

In the empty page I see a lot of resources loaded successfully and no error accrued during loading.
I can not see any reasonable error in nginx or keycloak logs that helps me why there is no visible data on loaded page.

But without any change to my keycloak or nging settings and just changing keycloak container image to 17.0.0 I receive page with data.

Has anybody a successful experience with version 18 or 19 and nginx proxy?
If so I really appreciate if you share some key points of your config for keycloak or nginx that help me follow your success.

Thanks a lot,
Ali 

![image](https://user-images.githubusercontent.com/5300198/183487452-0cae8a8e-dd91-433c-9cd2-39672e131e16.png)

![image](https://user-images.githubusercontent.com/5300198/183487831-bbb72a32-0ea6-4d26-84f2-3145a0f6c0ba.png)

**Comment by ayoubAnbara** [comment]
Hi, 
I ran Keycloak with nginx, this is my config:
`docker run -it -d --restart "unless-stopped" -p 9090:8080 -e PROXY_ADDRESS_FORWARDING="true"  -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin --name keycloak19  quay.io/keycloak/keycloak:19.0.1  [start-dev|start-prod] --proxy edge --http-relative-path /my-keycloak`

nginx:

```
location  /my-keycloak/ {

            proxy_pass http://localhost:9090/my-keycloak;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "upgrade";
            proxy_set_header Host $http_host;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-Host $host;
            proxy_set_header X-Forwarded-Server $host;
            proxy_set_header X-Forwarded-Port $server_port;
            proxy_set_header X-Forwarded-Proto $scheme;
            proxy_http_version 1.1;
            proxy_buffer_size   128k;
            proxy_buffers   4 256k;
            proxy_busy_buffers_size   256k;
            client_max_body_size 16m;
        }

```

**Reply by Ali-Javanmardi** [reply]
@ayoubAnbara , thanks for your help.
Although I could not solve my problem exactly by your sample but your confirmation about running KC behind Nginx helped me look more in my configuration again and find the problem point.

**Comment by Ali-Javanmardi** [comment, marked as answer]
Finally I could mange this proxy to run.

I found I have been set KC_HOSTNAME_PORT=443 in my docker compose file.
I found by removing this env variable it works as expected.

In the original documentation it was not clear for me how to set this variable.
Thanks
